### PR TITLE
Add unit tests based on samples from specification 3341 and 3342

### PR DESCRIPTION
**Pull request by MarcAntoine-Arnaud**
TODO:
* seq-3341-6-6channels-WAVEEX-16bit.wav: setIntegratedLoudness( -23.0f )
* seq-3341-13-[1-21]-24bits.wav: setMomentaryLoudness( -23.0f )
* seq-3341-14-24bit.wav: setMomentaryLoudness( -19.0f )

### CODE REVIEW

File: test/loudness-analyser.cpp [hunk 1]
```diff
@@ -312,7 +331,7 @@ TEST_P(CaseLoudnessAnalysis, Test)
     }
     if( ! isnan( _cfg.getIntegratedRange() ) )
     {
-        ASSERT_NEAR( _loudness.getIntegratedRange(), _cfg.getIntegratedRange(), 0.1 );
+        ASSERT_NEAR( _loudness.getIntegratedRange(), _cfg.getIntegratedRange(), 1.0 );
```
Comment (cchampet) [inline test/loudness-analyser.cpp]: All of the tests return a bad value for integrated range?

**Comment by cchampet** [comment]
From Travis:
```
[  PASSED  ] 69 tests.
```
:clap:

File: test/loudness-analyser.cpp [hunk 2]
```diff
@@ -6,72 +6,354 @@
...
+TEST_P(CaseLoudnessAnalysis, Test)
 {
-    checkEBUR128Analysis("1kHz Sine -40 LUFS-16bit.wav", -40, -40, -40);
+    if( ! isnan( _cfg.getIntegratedLoudness() ) )
+    {
+        ASSERT_NEAR( _loudness.getIntegratedLoudness(), _cfg.getIntegratedLoudness(), 0.1 );
+    }
+    if( ! isnan( _cfg.getIntegratedRange() ) )
+    {
+        ASSERT_NEAR( _loudness.getIntegratedRange(), _cfg.getIntegratedRange(), 1.0 );
```
Comment (cchampet) [inline test/loudness-analyser.cpp]: Can you explain this gap (1.0) between the expected and the current values?

File: test/loudness-analyser.cpp [hunk 3]
```diff
@@ -6,72 +6,354 @@
...
+    {
+        ASSERT_NEAR( _loudness.getMomentaryLoudness(), _cfg.getMomentaryLoudness(), 0.1 );
+    }
+    if( ! isnan( _cfg.getTruePeakValue() ) )
+    {
+        ASSERT_NEAR( _loudness.getTruePeakValue(), _cfg.getTruePeakValue(), 0.1 );
+    }
+    if( ! isnan( _cfg.getTruePeakInDbTP() ) )
+    {
+        ASSERT_GT( _loudness.getTruePeakInDbTP(), _cfg.getTruePeakInDbTP() - 0.4 );
```
Comment (cchampet) [inline test/loudness-analyser.cpp]: Can you explain those numbers (-0.4 and +0.2)?

File: test/loudness-analyser.cpp (hunk 2 above)
Comment (MarcAntoine-Arnaud) [inline test/loudness-analyser.cpp]: it's limits defined in EBU specifications : https://tech.ebu.ch/docs/tech/tech3342.pdf
LRA is valid when the system can found the value around +-1 LU around the (LRA = 10 ±1
LU)

File: test/loudness-analyser.cpp (hunk 3 above)
Comment (MarcAntoine-Arnaud) [inline test/loudness-analyser.cpp]: it's limits defined in EBU specifications : https://tech.ebu.ch/docs/tech/tech3341.pdf
True peak can be found with this approximation.

File: test/loudness-analyser.cpp (hunk 2 above)
Comment (cchampet) [inline test/loudness-analyser.cpp]: Indeed. I thought first that there was no tolerance for these expected values. And so they could be used to calibrate our algorithm and check if it returns the exact value. But the tables of papers 3341-3342 describe for each test an "Expected response and accepted tolerances", as you said.
LGTM!

**Comment by cchampet** [comment]
See PR #56

**Pull request closed without merging**